.github/workflows/pr-breaking-change-check.yaml

```diff
@@ -0,0 +1,49 @@
+name: Breaking Change Check
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'FFXIVClientStructs/**/*.cs'
+  
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.base_ref }}
+    - name: Install dependencies for main branch
+      run: dotnet restore
+    - name: Build main branch
+      run: dotnet build FFXIVClientStructs/FFXIVClientStructs.csproj --output ida/cs-main/
+    - name: Switch to PR branch
+      run: git checkout ${{ github.head_ref }}
+    - name: Install dependencies for PR branch
+      run: dotnet restore
+    - name: Build PR branch
+      run: dotnet build FFXIVClientStructs/FFXIVClientStructs.csproj --output ida/cs-pr/
+    - name: Check for breaking changes
+      id: breaking_changes
+      run: |
+        dotnet tool install -g Microsoft.DotNet.ApiCompat.Tool
+        testout=$(apicompat -l ida/cs-main/FFXIVClientStructs.dll -r ida/cs-pr/FFXIVClientStructs.dll)
```

Comment:
Problems are pushed to STDERR, success message (if any) is pushed to STDOUT. This is specifically why we are testing the `"APICompat ran successfully without finding any breaking changes."` string, at least on a Windows runner.